### ISSUE TITLE
How to disable the daily updates on snap and just update when needed?

### ISSUE THREAD

**Issue by therampagerado** [wekan/wekan#2120]
I'm having the following issue where the system hangs once a day (usually right when we're have most of our work around 3-4 PM). I think it's updating then and can't start some of it's services.

How can I disable the daily snap updates and just update once a week or in two weeks. This can be done manually, no need for automation.

Thanks!

**Comment by xet7** [comment]
@therampagerado 

You can set at what time Wekan etc snap packages are updated:
https://github.com/wekan/wekan-snap/wiki/Install#5-install-all-snap-updates-automatically-between-0200am-and-0400am

For example:
```
snap set core refresh.schedule=02:00-04:00
```
So set that to your nighttime.

If you want to update manually, then you change from Snap version to Docker version. See Wekan GitHub wiki Backup page for import/export info.

**Comment by cherio** [comment]
It is a shame there is no good solution for disabling snap auto updates. The pain I feel is like the one when Windows forces an update on you and there is not much you can do.

One hack I use is disable access to the update site in `/etc/hosts` by adding a line
`127.0.0.1 api.snapcraft.io`

An even uglier solution is adding a rule in IP tables dropping packets designated to `api.snapcraft.io`.

It is truly a sad subject and it's been this way for years.

**Comment by xet7** [comment]
@cherio 

It's a shame many are running old vulnerable Wekan versions.

**Comment by xet7** [comment]
@cherio 

With Docker and [Prebuilt Bundle](https://github.com/wekan/wekan/wiki/Raspberry-Pi) installs it's possible to run any Wekan version, those do not have automatic update.

**Comment by xet7** [comment]
@cherio 

Really, security researchers are asking me when most of Wekan users have updated to new enough Wekan version where critical vulnerabilities have been fixed, so that I would be confortable enough to publish full details of those vulnerabilities with PoC (Proof of Concept) code how to hack Wekan, and apply for CVE numbers. Really, I would not like Wekan users to get hacked. Currently when CVEs are published, usually within 3 days hackers are scanning the Internet for vulnerable versions and exploiting them. For more info about current state of security, [listed to this SaltStack The Hacks podcast](https://www.saltstack.com/the-hacks/).

**Comment by cherio** [comment]
@xet7 I think you mispercieved my answer. When answering the question (you've got to agree I offered a concrete answer to the question) my criticism was not directed at wekan but at the way snapd performs updates and gives no good way of doing it manually

**Comment by xet7** [comment]
@cherio 

Yeah, nice way for you to [continue your trolling](https://github.com/wekan/wekan/wiki/FAQ#why-am-i-called-a-troll).

**Comment by xet7** [comment]
Snap is the best automatic update system ever designed for mankind. It updates fast and automatically. It is huge win when there is new security fix for Wekan.

**Comment by cherio** [comment]
Holy cow, you just updated the wiki page by adding half of my post there and "making" me fit a troll profile. I don't even know how to call this sort of behavior. That's beyond civilized.

My post must have hurt you somehow even though it is a direct answer to the question offering a working solution and not trying to steal anyone's thunder. You simply didn't like it for whatever personal reasons.

**Comment by xet7** [comment]
It's possible to [schedule Snap updates to happen at night](https://github.com/wekan/wekan-snap/wiki/Install#5-install-all-snap-updates-automatically-between-0200am-and-0400am).

**Comment by cherio** [comment]
> Please describe why having automatic updates for Wekan would be a pain?

Why should I? I never stated it's a pain to do so "_for Weken_" in the first place. You should read my answer again and if you see me talking about wekan there then you sir either have paranoia or twist meaning for the sake of argument. I answered the question "How can I disable the daily snap updates and just update once a week or in two weeks. **This can be done manually, no need for automation.**". Both the question and my answer are about snap updates.

You don't like my answer I get it. So leave a comment that you don't recommend and can't sign off on that approach.

Instead you are trying to discredit me in a unique way. You declare me a troll and reference a freshly updated project wiki page with a personal definition of a troll partly shaped around my post.

This is sick.

**Comment by xet7** [comment]
@cherio 

It's already possible to update once a week, once a month, or until specific day with Snap.
https://snapcraft.io/docs/keeping-snaps-up-to-date

No need for hacks.

I'm only against turning off updates permanently.

**Comment by xet7** [comment]
@cherio 

Now that I re-read your original comment, the solution you provided works for turning off automatic updates. I think there is currently no other way to permanently stop updates. This could be useful when running Wekan behind firewall.

I have added this info to [Snap install page](https://github.com/wekan/wekan-snap/wiki/Install) and new [Automatic update schedule page](https://github.com/wekan/wekan-snap/wiki/Automatic-update-schedule).

While thinking about this, I do remember, that in Linux it's possible to have automatic updates enabled or disabled.

I apologize for the inconvenience and misunderstanding after coming to conclusion that while you used troll words in your original comment, unfortunately you are not a troll.

It seems that you are right, and I just did not get it.

So tonights troll party is cancelled. Thats all folks, and see you all in the next party!
